src/components/TodoModal/TodoModal.tsx

```diff
@@ -15,25 +29,36 @@ export const TodoModal: React.FC = () => {
               className="modal-card-title has-text-weight-medium"
               data-cy="modal-header"
             >
-              Todo #2
+              Todo #{todo.id}
             </div>
 
             {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
```

Comment:
The ESLint disable comment suggests that the close button lacks an associated label. Consider adding an `aria-label` to the button for better accessibility.